daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Error.scala

```diff
@@ -45,6 +45,11 @@ final case class ContractNotFound(ci: ContractId) extends Error {
   override def detailMsg: String = msg
 }
 
+final case class ContractKeyNotFound(key: GlobalKey) extends Error {
+  override def msg = s"dependency error: couldn't find key: $key"
```

Comment:
I kept the error message the same to avoid unnecessarily breaking things like the ledger API test tool.